### ISSUE TITLE
Roslyn analyzer: "does not have accessible to set from the same dll. It cannot support to inject by the source generator."

### ISSUE THREAD

**Issue by RedGlow** [hadashiA/VContainer#730]
Hello,
  I'm trying to use the Roslyn [source generator approach](https://vcontainer.hadashikick.jp/optimization/source-generator#2-mark-as-roslynanalyzer) on a Unity 6 project.

  I've started from a minimal project that contains the following definitions:

_Service.cs_
```csharp
namespace Test
{
    public class Service
    {
        public string Hello()
        {
            return "Hello";
        }
    }
}
```

_View.cs_
```csharp
using UnityEngine;
using VContainer;

namespace Test
{
    public class View: MonoBehaviour
    {
        [Inject] public Service Service { get; set; }

        private void Start()
        {
            Debug.Log(Service.Hello());
        }
    }
}
```

_GameLifetimeScope.cs_
```csharp
using Test;
using VContainer;
using VContainer.Unity;

public class GameLifetimeScope : LifetimeScope
{
    protected override void Configure(IContainerBuilder builder)
    {
        builder.Register<Service>(Lifetime.Scoped);
    }
}
```

And a simple hierarchy:
![image](https://github.com/user-attachments/assets/6b0e691c-ba60-4f27-b209-181b748453f5)
![image](https://github.com/user-attachments/assets/276b4d07-f14b-4c54-9183-4420511d96d3)

With this setup, the view correctly prints the message.

From my understanding, I could remove the auto-inject of the View game object and use source generation. I followed the instructions above, but then I get this compilation error:

```
Assets\Test\View.cs(8,33): warning VCON0008: The [Inject] 'Service' does not have accessible to set from the same dll. It cannot support to inject by the source generator.
```

It's not really clear to me what this phrase means. Looking for this error in the source code I found [this snippet](https://github.com/hadashiA/VContainer/blob/7d839caeb73b6b97d996688333e26ed534f55aad/VContainer.SourceGenerator/Emitter.cs#L135C1-L143C1):
```csharp
if (propSymbol.SetMethod == null || !propSymbol.SetMethod.IsInitOnly || !propSymbol.SetMethod.CanBeCallFromInternal())
{
    context.ReportDiagnostic(Diagnostic.Create(
        DiagnosticDescriptors.PrivatePropertyNotSupported,
        propSymbol.Locations.FirstOrDefault() ?? typeMeta.GetLocation(),
        propSymbol.Name));
    error = true;
}
```

Which seems to indicate that the following conditions must be met:
- the property must have a setter
- the setter must not be an init (maybe?)
- and must have at least an "internal" level

That "init" part made me curious, so I changed the definition of the property from `{get; set;}` to `{get; init;}`, by adding the IsExternalInit which is missing [according to Unity documentation](https://docs.unity3d.com/6000.1/Documentation/Manual/csharp-compiler.html), and at that point the source generator actually produced the code (even though the code seems to indicate the contrary), but then it obviously failed the compilation with the error `VContainer.SourceGenerator\VContainer.SourceGenerator.VContainerIncrementalSourceGenerator\Test.ViewGeneratedInjector.g.cs(19,14): error CS8852: Init-only property or indexer 'View.Service' can only be assigned in an object initializer, or on 'this' or 'base' in an instance constructor or an 'init' accessor.` because it needs a setter:

```csharp
__x.Service = (global::Test.Service)resolver.ResolveOrParameter(typeof(global::Test.Service), "Service", parameters);
```

My suspect is that the IsInitOnly property is wrong in Unity 6, but I didn't dig any further in the code.

You can find my test project here: https://github.com/RedGlow/vcontainer-bug-report .

**Comment by hadashiA** [comment]
Thank you for your detailed report.

After checking, it seems that there was my mistake in the conditional expression in the source code you provided.

I've confirmed that the problem has been solved with the fix for #738, so I've merged it.
We'll release it soon.

**Comment by hadashiA** [comment]
I've released a fix.
https://github.com/hadashiA/VContainer/releases/tag/1.16.8
